apis/network/v1alpha3/resourcerecordset_types.go

```diff
@@ -0,0 +1,149 @@
+
+	// DNSRecordTypePtr represents DNS PTR record type.
+	DNSRecordTypePtr DNSRecordType = "PTR"
+
+	// DNSRecordTypeSrv represents DNS SRV record type.
+	DNSRecordTypeSrv DNSRecordType = "SRV"
+
+	// DNSRecordTypeSpf represents DNS SPF record type.
+	DNSRecordTypeSpf DNSRecordType = "SPF"
+
+	// DNSRecordTypeAaaa represents DNS AAAA record type.
+	DNSRecordTypeAaaa DNSRecordType = "AAAA"
+
+	// DNSRecordTypeCaa represents DNS CAA record type.
+	DNSRecordTypeCaa DNSRecordType = "CAA"
+)
+
+// ChangeAction defines the valid actions that can be performed on a ResourceRecordSet.
+type ChangeAction string
+
+const (
+	// ChangeActionCreate represents a Resource Record CREATE operation.
+	ChangeActionCreate ChangeAction = "CREATE"
+
+	// ChangeActionDelete represents a Resource Record DELETE operation.
+	ChangeActionDelete ChangeAction = "DELETE"
+
+	// ChangeActionUpsert represents a Resource Record UPSERT operation.
+	ChangeActionUpsert ChangeAction = "UPSERT"
+)
+
+// +kubebuilder:object:root=true
+
+// ResourceRecordSet is a managed resource that represents an AWS Route53 Resource Record.
+// +kubebuilder:printcolumn:name="NAME",type="string",JSONPath=".status.forProvider.name"
+// +kubebuilder:printcolumn:name="TYPE",type="string",JSONPath=".status.atProvider.Type"
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type ResourceRecordSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ResourceRecordSetSpec   `json:"spec"`
+	Status ResourceRecordSetStatus `json:"status,omitempty"`
+}
+
+// ResourceRecordSetSpec defines the desired state of an AWS Route53 Resource Record.
+type ResourceRecordSetSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ResourceRecordSetParameters `json:"forProvider"`
+}
+
+// ResourceRecordSetStatus represents the observed state of a ResourceRecordSet.
+type ResourceRecordSetStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     ResourceRecordSetObservation `json:"atProvider"`
+}
+
+// ResourceRecord holds the DNS value to be used for the record.
+type ResourceRecord struct {
+	// The current or new DNS record value, not to exceed 4,000 characters. In the
+	// case of a DELETE action, if the current value does not match the actual value,
+	// an error is returned.
+	Value *string `json:"Value"`
+}
+
+// ResourceRecordSetParameters define the desired state of an AWS Route53 Resource Record.
+type ResourceRecordSetParameters struct {
+	// Name of the record that you want to create, update, or delete.
+	Name *string `json:"name"`
```

Comment:
It seems like this name is the unique identifier of the record set? If so I think we should remove this field and derive the name from the CR's external name annotation using `meta.GetExternalName`.

Reply:
I believe that from instead of asking the user to provide the name under 

```
metadata:
  annotation: crossplane.io/external-name:
```
it would be easier from a UX perspective to let them specify that in 

```
spec:
  forProvider:
    name:
```

and then set the annotation internally, as that way it also follows the struct design as specified by the AWS SDK.